Mixpanel/Mixpanel.m

```diff
@@ -49,10 +49,33 @@ + (Mixpanel *)sharedInstanceWithToken:(NSString *)apiToken
 
 + (Mixpanel *)sharedInstance
 {
-    if (sharedInstance == nil) {
-        MPLogWarning(@"sharedInstance called before sharedInstanceWithToken:");
+    if (!instances || instances.count == 0) {
+        MPLogWarning(@"sharedInstance called before creating a Mixpanel instance");
+        return nil;
+    }
+
+    if (instances.count > 1) {
+        MPLogWarning(@"sharedInstance can only be called when you have one Mixpanel instance");
+        return nil;
+    }
+
+    return [instances objectForKey:[[instances keyEnumerator] nextObject]];
```

Comment:
since there's only one i think `[[instances allValues] firstObject]` is clearer and has the same performance